src/app/services/favourites.service.ts

```diff
@@ -0,0 +1,90 @@
+import { Injectable, inject } from '@angular/core';
+import { Firestore, doc, getDoc, setDoc, arrayUnion, arrayRemove, updateDoc, DocumentReference, DocumentData } from '@angular/fire/firestore';
+import { User } from '@angular/fire/auth';
+import { AuthenticationService } from './authentication.service';
+import { Activity } from '../model/activity';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class FavouritesService {
+  private firestore: Firestore = inject(Firestore);
+  private currentUser: User | null;
+  private userDocRef: DocumentReference<DocumentData>;
+
+  constructor(private authService: AuthenticationService) {
+    this.setupAuthStatusListener();
+  }
+
+  private setupAuthStatusListener() {
+    // Subscribe to the authStatusSub BehaviorSubject in the AuthenticationService
+    this.authService.currentAuthStatus.subscribe((user) => {
+      this.currentUser = user;
+
+      if (user) {
+        this.userDocRef = doc(this.firestore, 'users', user.uid);
+      }
+    });
+    
+  }
+
+  async isActivityFavorited(activityId: string | null): Promise<boolean> {
+    if (!this.currentUser && !activityId) {
+      throw new Error('User not logged in or activityId is null');
+    }
+
+    const userDocSnapshot = await getDoc(this.userDocRef);
+    const favorites = userDocSnapshot.get('favorites') || [];
```

Comment:
Can we have this array in memory instead of making a request for every activity?

Reply:
If you have `favorites` array as a property, load it in constractor and update in add/remove functions?